code/frameworks/angular/src/client/angular-beta/utils/PropertyExtractor.ts

```diff
@@ -0,0 +1,123 @@
+import { CommonModule } from '@angular/common';
+import { NgModuleMeta } from '@angular/compiler-cli/src/ngtsc/metadata';
+import { NgModule, Provider } from '@angular/core';
+import { NgModuleMetadata } from '../../types';
+import { isDeclarable, isStandaloneComponent } from './NgComponentAnalyzer';
+import { isComponentAlreadyDeclared } from './NgModulesAnalyzer';
+
+/**
+ * Flattens Array Deeply
+ *
+ * - Executes recursively
```

Comment:
Also, let's remove comments for self-explanatory and self-explaining functions. In this case, the name of the function is self-explaining. Let's only add documentation if it is indispensable.

Reply:
👍